### ISSUE TITLE
Is there are some ways to know on which subsample of data the XGBoost tree was fitted?

### ISSUE THREAD

**Issue by PowderL** [dmlc/xgboost#7257]
I am working on data science with xgboost. Recently, I want to diagnose the model with the self-defined method.  And I want to know if there exists any way to get the subset of data sampled from training data for each xgboost tree.

**Comment by trivialfis** [comment]
Unfortunately, the information is not available to the users.

**Comment by trivialfis** [comment]
It's actually straightforward to implement the sampling yourself.  Just define a custom objective and set the hessian value of not selected samples to 0.

**Comment by PowderL** [comment]
> It's actually straightforward to implement the sampling yourself. Just define a custom objective and set the hessian value of not selected samples to 0.

OK, thanks so much, I really appreciate for your reply.

**Comment by PowderL** [comment]
> It's actually straightforward to implement the sampling yourself. Just define a custom objective and set the hessian value of not selected samples to 0.

OK, thanks so much, I really appreciate for your reply. And, is there any way to get the random seed like the  random state in the numpy package to sample the train data for each xgboost tree?

**Comment by PowderL** [comment]
> I am working on data science with xgboost. Recently, I want to diagnose the model with the self-defined method. And I want to know if there exists any way to get the subset of data sampled from training data for each xgboost tree.
